From 8ad81dfd6a44b079b7a5c5752c40a3c5c47fea00 Mon Sep 17 00:00:00 2001
From: Frederik Carlier <frederik.carlier@keysight.com>
Date: Thu, 5 Sep 2024 14:43:01 -0700
Subject: [PATCH] Mark PACKAGE_SCOPE as public on Windows

---
 Headers/AppKit/AppKitDefines.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Headers/AppKit/AppKitDefines.h b/Headers/AppKit/AppKitDefines.h
index 54f6661eb..e77461afb 100644
--- a/Headers/AppKit/AppKitDefines.h
+++ b/Headers/AppKit/AppKitDefines.h
@@ -72,7 +72,7 @@
 
 #endif
 
-#if defined(__clang__) && !defined(__MINGW32__)
+#if defined(__clang__) && !defined(__MINGW32__) && !defined(WIN32)
 #  define PACKAGE_SCOPE @package
 #else
 #  define PACKAGE_SCOPE @public
-- 
2.45.2.windows.1