From d46884d69445d4e3c4f4d6d43e5e7cca42f7e55c Mon Sep 17 00:00:00 2001
From: Frederik Carlier <frederik.carlier@keysight.com>
Date: Fri, 6 Sep 2024 15:50:28 +0000
Subject: [PATCH] Disable building Tools

---
 GNUmakefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/GNUmakefile b/GNUmakefile
index 97e456771..4abc0efa7 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -56,7 +56,6 @@ Source \
 Images \
 Sounds \
 Model \
-Tools \
 Panels \
 PrinterTypes \
 TextConverters \
-- 
2.46.0.windows.1