From ec6299f72faa04f4bb019d4f2a0984fe832b09c0 Mon Sep 17 00:00:00 2001
From: ethanc8 <echaroenpitaks@gmail.com>
Date: Sun, 22 Jun 2025 13:31:44 -0500
Subject: [PATCH] expose declarations in NSDebug.h even when NDEBUG is defined

---
 Headers/Foundation/NSDebug.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/Headers/Foundation/NSDebug.h b/Headers/Foundation/NSDebug.h
index dd0ad1153..29cc565b0 100644
--- a/Headers/Foundation/NSDebug.h
+++ b/Headers/Foundation/NSDebug.h
@@ -115,8 +115,6 @@ extern "C" {
  */
 extern BOOL NSDebugEnabled;
 
-#ifndef	NDEBUG
-
 /**
  * Used internally by NSAllocateObject() ... you probably don't need this.
  */
@@ -321,8 +319,6 @@ GS_EXPORT void  GSSetDebugAllocationFunctions(
   void (*newAddObjectFunc)(Class c, id o),
   void (*newRemoveObjectFunc)(Class c, id o));
 
-#endif
-
 /**
  * Enable/disable zombies.
  * <p>When an object is deallocated, its isa pointer is normally modified
