From a4a9d9d2fa6419a53eb9c48c176918cf4888b4b2 Mon Sep 17 00:00:00 2001
From: Frederik Carlier <frederik.carlier@keysight.com>
Date: Mon, 9 Sep 2024 09:13:16 +0000
Subject: [PATCH] Don't link with libm on Windows

---
 configure    | 2 +-
 configure.ac | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 55ce041..b0eaaa5 100755
--- a/configure
+++ b/configure
@@ -6375,7 +6375,7 @@ else $as_nop
 fi
 
 if test "$have_gdi32" = yes; then
-  WIN32_LIBS="-lgdi32 -lm"
+  WIN32_LIBS="-lgdi32"
 fi
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for main in -lmsimg32" >&5
 printf %s "checking for main in -lmsimg32... " >&6; }
diff --git a/configure.ac b/configure.ac
index 93b8ba1..2b73c86 100644
--- a/configure.ac
+++ b/configure.ac
@@ -393,7 +393,7 @@ AC_CHECK_LIB(X11, XInternAtoms,
 #--------------------------------------------------------------------
 AC_CHECK_LIB(gdi32, main, have_gdi32=yes, have_gdi32=no)
 if test "$have_gdi32" = yes; then
-  WIN32_LIBS="-lgdi32 -lm"
+  WIN32_LIBS="-lgdi32"
 fi
 AC_CHECK_LIB(msimg32, main, have_msimg32=yes, have_msimg32=no)
 if test "$have_msimg32" = yes; then
-- 
2.46.0.windows.1

